include/stdx/atomic.hpp

```diff
@@ -0,0 +1,126 @@
+#pragma once
+
+#include <conc/atomic.hpp>
+
+#include <atomic>
+#include <type_traits>
+
+#if __cplusplus >= 202002L
+#define CPP20(...) __VA_ARGS__
+#else
+#define CPP20(...)
+#endif
+
+namespace stdx {
+inline namespace v1 {
+template <typename T> class atomic {
+    static_assert(std::is_trivially_copyable_v<T> and
+                      std::is_copy_constructible_v<T> and
+                      std::is_move_constructible_v<T> and
+                      std::is_copy_assignable_v<T> and
+                      std::is_move_assignable_v<T>,
+                  "Atomic values must be trivially copyable, copy "
+                  "constructible and copy assignable");
+
+    using elem_t = ::atomic::atomic_type_t<T>;
+    constexpr static auto alignment = ::atomic::alignment_of<T>;
+
+    static_assert(std::is_convertible_v<elem_t, T>,
+                  "::atomic::atomic_type is specialized, but the type it "
+                  "produces is not convertible to T");
+    static_assert(std::is_convertible_v<T, elem_t>,
+                  "::atomic::atomic_type is specialized, but the type it "
+                  "produces is not convertible from T");
+
+    alignas(alignment) elem_t value;
+
+  public:
+    using value_type = T;
+
+    constexpr atomic()
+#if __cplusplus >= 202002L
+        requires std::is_default_constructible_v<elem_t>
+#endif
+        : value{} {
+    }
+    constexpr atomic(T t) : value{static_cast<elem_t>(t)} {}
+    atomic(atomic const &) = delete;
+    auto operator=(atomic const &) -> atomic & = delete;
+
+    [[nodiscard]] auto
+    load(std::memory_order mo = std::memory_order_seq_cst) const -> T {
+        return static_cast<T>(::atomic::load(value, mo));
+    }
+
+    void store(T t, std::memory_order mo = std::memory_order_seq_cst) {
+        ::atomic::store(value, static_cast<elem_t>(t), mo);
+    }
+
+    operator T() const { return load(); }
```

Comment:
Hmm, OK. The only situation it would affect is an explicit cast being unused. because an implicit cast wouldn't happen unless it were used I guess. Can't hurt?